### DISCUSSION TITLE
Capture bed variation output?

### DISCUSSION THREAD

**Discussion by beffjessen** [jneilliii/OctoPrint-BedLevelVisualizer#592] [Q&A]
Not sure if this is the right place to ask...

Once you tram the bed and run bed leveling, you're ready to start printing; but at some point the bed will drift out of parallel to the point that it needs to be manually trammed again. As I understand it, there's a certain tolerance, below which ABL can reasonably correct the rest of the variation (FWIW, I shoot for a maximum variation of <1mm). 

It would be awesome if you could run ABL every N prints, and then flag once the variation has drifted outside of a user-defined tolerance (maybe by running a custom command with the TopTemp plugin?). Is this something that can already be done with an existing plugin? Or is there a way to capture the `variation` output from the BLV plugin (or even straight from EEPROM) as input for a separate script? I appreciate any help!

**Comment by jneilliii** [comment]
You should check out that new plugin [SmartABL](https://plugins.octoprint.org/plugins/SmartABL/). It does exactly what you mention relative to only running every n prints, but don't think it captures variance or the like but it might.
